### PR TITLE
nixos home-assistant: a couple of fixes

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

a) set path to /run/wrappers so ping works (what gets added is /run/wrappers/{bin,sbin})
b) restart if the configuration file changes
c) run via a target so we can easily inject other components (config copier, appdaemon)

Cc: @dotlambda 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/misc/home-assistant.nix [hunk 1]
```diff
@@ -121,6 +120,19 @@ in {
         ReadWritePaths = "${cfg.configDir}";
         PrivateTmp = true;
       };
+      restartTriggers = [
+        configFile
+      ];
+      path = [
+        "/run/wrappers"
```
Comment (Mic92) [inline nixos/modules/services/misc/home-assistant.nix]: Can you mention `ping` here?

File: nixos/modules/services/misc/home-assistant.nix [hunk 2]
```diff
@@ -121,6 +120,19 @@ in {
         ReadWritePaths = "${cfg.configDir}";
         PrivateTmp = true;
       };
+      restartTriggers = [
+        configFile
```
Comment (dotlambda) [inline nixos/modules/services/misc/home-assistant.nix]: Good catch!

File: nixos/modules/services/misc/home-assistant.nix (hunk 1 above)
Comment (peterhoeg) [reply nixos/modules/services/misc/home-assistant.nix]: done

**Comment by dotlambda** [comment]
Can you rebase on staging so we can run the test?

**Comment by peterhoeg** [comment]
I've run the test locally against unstable (with the pytest-mock upgrade reverted), which was fine, but of course we can try against staging.

**Comment by peterhoeg** [comment]
Ehhh...... What the what? Rebasing against staging didn't exactly go as planned...

**Comment by dotlambda** [comment]
@peterhoeg I have rebased. We can wait with merging until the pytest-mock fix is in master.

@GrahamcOfBorg test home-assistant

File: nixos/modules/services/misc/home-assistant.nix (hunk 2 above)
Comment (dotlambda) [reply nixos/modules/services/misc/home-assistant.nix]: Actually, thinking about it, shouldn't this be unnecessary becasue `configFile` is mentioned in `preStart`?

Comment (Mic92) [reply nixos/modules/services/misc/home-assistant.nix]: Yes. It should be unnecessary, since `preStart` is regenerated if `cfg.config` changes.

**Comment by dotlambda** [comment]
I don't know why ofborg failed. I tested locally and it works fine.

**Comment by dotlambda** [comment]
@GrahamcOfBorg test home-assistant

**Comment by dotlambda** [comment]
@GrahamcOfBorg test home-assistant

**Comment by dotlambda** [comment]
@peterhoeg Feel free to backport to 18.03.

**Pull request merged**